### DISCUSSION TITLE
Prediction-Asssited Labelling Consistantly Mislabelling Anchor Point

### DISCUSSION THREAD

**Discussion by mollywestbrook** [talmolab/sleap#1177] [Help!] [answered]
Hi all! Trying out SLEAP on some swimming cichlids. The model has 107 frames so far, and I'm finding that the model is able to predict the fish pretty well (>95% of the time, usually) but continually misplaces 2 points, the "head" point (which is designated the anchor point) and the following skeletal point (spine 1). All other points are usually spot on. A picture of one of my labelled fish, overlaid with predictions:
![image](https://user-images.githubusercontent.com/48096260/218770694-9a7697b7-c129-4e1f-9aeb-d0bbb86b3a62.png)
and a predicted instance:
![image](https://user-images.githubusercontent.com/48096260/218771503-25f48752-9d52-4be9-b0a3-f2dce6d073b7.png)
I've run the model 7 times now. I began with 20 random labelled frames. Each time I label 20 more random frames, go and correct those predictions, rerun the model including the new and original frames, spit out 20 predictions again, etc. Yet each time the predicted frames consistantly mislabel the two points above no matter how many times I've done this. 

Model settings are all standard settings (multi-animal top-down style model, 2.5 sigma for centroids and nodes, best model and visualize predictions enabled, predicting on 20 random frames, screen grabs of the centroid and centered instance settings:
![image](https://user-images.githubusercontent.com/48096260/218772024-eea11b32-3b0c-4aab-9672-e23661262c9f.png)
![image](https://user-images.githubusercontent.com/48096260/218772059-1beb1e89-bd3d-4d26-94ce-9dc13b7912e7.png)

Is it typical for the model to mislabel the anchor point? Should I be giving the model frames from another video? Should I relabel more frames from the labeling suggestions tab? Any guidance would be great. Thanks!

**Comment by roomrys** [comment, marked as answer]
Hi @mollywestbrook,

In both images you provided, the head and anchor points are marked as not visible (we can tell because they have a gray box around the node label). If this schema (of marking the head and spine1 as not visible) is consistent throughout your labels, then we would expect this behavior in the predictions. You can toggle node visibility by right clicking on the node.

Let us know if this helps!

Thanks,
Liezl

P.S. You can loop through your user-labeled frames using `Ctrl + U` and quickly double check for any gray node label boxes which should not be there.

**Reply by mollywestbrook** [reply]
Yes, the problem is solved:
![image](https://user-images.githubusercontent.com/48096260/219093104-dcfbf4dc-cd33-4189-873f-a7f5dba99539.png)

Out of curiosity, because it seems that the first instance I marked I happened to mark those two points as not visible, would that propagate to every new instance of the skeleton? I just am curious how this may have happened. Thanks for your guidance.

**Reply by roomrys** [reply]
Hi @mollywestbrook,

Glad it is solved!

The default for adding a new instance in the GUI is to copy the most recently created instance. This is likely how these non visible nodes propagated throughout the project.

Thanks,
Liezl
